python_client/web_api_client.py

```diff
@@ -0,0 +1,135 @@
+import argparse
+
+import requests
+from datetime import datetime
+from dataclasses import dataclass
+import time
+
+# Put here your default path to the .pptx file
+DEFAULT_FILE_PATH = "C:\\Users\\User\\Desktop\\excellenteam\\excellenteam_python\\t.pptx"
+
+
+@dataclass
+class Status:
+    """
+    Represents the status of a file processing task.
+
+    Attributes:
+        status (str): The current status of the processing file.
+        filename (str): The name of the file being processed.
+        timestamp (datetime): The timestamp when the processing started.
+        explanation (str): The presentation's explanation from chat gpt.
+    """
+    status: str
+    filename: str
+    timestamp: datetime
+    explanation: str
+
+    def is_done(self):
+        """Check if the file processing is completed."""
+        return self.status == 'done'
+
+
+class WebAppClient:
+    """
+    A client for interacting with a web application to upload files and check their processing status.
+
+    Attributes:
+        base_url (str): The base URL of the web application's API.
+    """
+
+    def __init__(self, base_url):
+        self.base_url = base_url
+
+    def upload(self, file_path):
+        """
+        Uploads a file to the web application.
+
+        Args:
+            file_path (str): The local file path to be uploaded.
+
+        Returns:
+            str: The unique identifier (UID) associated with the uploaded file.
+
+        Raises:
+            Exception: If an error occurs during the file upload.
+        """
+        url = f"{self.base_url}/upload"
+        files = {'file': open(file_path, 'rb')}
+        response = requests.post(url, files=files)
+
+        if response.status_code == 200:
+            data = response.json()
+            return data['uid']
+        else:
+            raise Exception(f"Error uploading file: {response.text}")
+
+    def status(self, uid):
+        """
+        Gets the processing status of a file.
+
+        Args:
+            uid (str): The unique identifier (UID) of the file whose status is to be checked.
+
+        Returns:
+            Status: An instance of the Status class representing the file's processing status.
+
+        Raises:
+            Exception: If an error occurs while getting the status.
+        """
+        url = f"{self.base_url}/status/{uid}"
+        response = requests.get(url)
+        if response.status_code == 200:
+            data = response.json()
+            status = data['status']
+            filename = data['filename']
+            timestamp = datetime.strptime(data['timestamp'][:-5], "%Y%m%d%H%M%S")
```

Comment:
Assign constants to descriptive variables to enhance readability and simplify future changes ^__^ . Directly using them can lead to confusion and maintenance challenges.